src/main/java/de/metas/ui/web/material/cockpit/MaterialCockpitRowRepository.java

```diff
@@ -57,10 +63,21 @@
 @Repository
 public class MaterialCockpitRowRepository
 {
-	private final transient CCache<Integer, List<I_M_Product>> orgIdToproducts = CCache.newCache(
-			I_M_Product.Table_Name + "#by#" + I_M_Product.COLUMNNAME_AD_Org_ID,
-			10, // initial size
-			CCache.EXPIREMINUTES_Never);
+	private static final String SYSCONFIG_EMPTY_PRODUCTS_LIMIT = "de.metas.ui.web.material.cockpit.MaterialCockpitRowRepository.EmptyProducts.Limit";
+	private static final int DEFAULT_EMPTY_PRODUCTS_LIMIT = 2000;
+
+	private static final String SYSCONFIG_EMPTY_PRODUCTS_CACHESIZE = "de.metas.ui.web.material.cockpit.MaterialCockpitRowRepository.EmptyProducts.CacheSize";
+	private static final int DEFAULT_EMPTY_PRODUCTS_CACHESIZE = 10;
+
+	private final transient CCache<CacheKey, List<I_M_Product>> productFilterVOToProducts;
```

Comment:
i think a list/set of ProductId(s) would be sufficient.

Reply:
checked, I think you are right! trying it

Reply:
done ✔️